latest/index.html

```diff
@@ -920,6 +940,60 @@ <h3>Data Extension Responses</h3>
         </p>
       </section>
     </section>
+    <section>
+      <h2>Security Considerations</h2>
+
+      <section>
+        <h3>Preventing Abuse</h3>
+        <ul>
+          <li>Reconciliation services SHOULD guard against too frequent requests to prevent [[DoS]] attacks (see <a href="#error-handling-and-rate-limiting">Error Handling and Rate-limiting</a>).</li>
+          <li>Reconciliation services SHOULD guard against malicious code when querying their underlying data store with data from <a>reconciliation queries</a> to avoid code [[injection]].</li>
+          <li>Reconciliation services SHOULD NOT parse the JSON requests sent by clients using a code execution mechanism like JavaScript's <code>eval()</code>, since requests could contain malicious code.</li>
+          <li>Reconciliation clients that display data from <a href="#preview-service">preview services</a> SHOULD ensure that the data is encoded, sanitized, or isolated to prevent [[XSS]] and [[CSRF]] attacks.</li>
+          <li>Reconciliation clients SHOULD provide a configuration option to disallow calls to localhost, with the default set to disallow, as a malicious service could cause the client to make arbitrary POST requests to itself.</li>
```

Comment:
Yeah, I also stopped there yesterday and had to think about it for a bit. So maybe it'd be good to elaborate in the specs, if we agree it makes sense and applies to us. I've got this from the [ActivityPub](https://www.w3.org/TR/activitypub/#security-localhost) and [Webmention](https://www.w3.org/TR/webmention/#avoid-sending-webmentions-to-localhost) specs, which both mention it. Maybe we could adapt and combine the explanations from these specs like this to make it clearer:

"If a reconciliation client has any services that listen on localhost and don't require authentication, it's possible for a malicious reconciliation service to provide URLs in its manifest that could cause the client to make arbitrary POST requests to itself. Therefore, if your reconciliation client does permit making requests to localhost URLs for development purposes, it SHOULD be a configuration option which defaults to off."

Reply:
Actually, which of the URLs provided in the manifest are used for POST requests? The suggest services use GET, so does the preview service, property proposal and so on… The only POST requests are made to the reconciliation endpoint itself (but that is not a URL that the manifest is able to define).

I am not saying this is by design (it is probably rather by luck), but as far as I can tell we are safe here. And it should only get better with #91 since manifests can no longer point to arbitrary URLs.